File: 2025-09-04/vowelRepeater.js
function repeatVowels(str) {
  const regex = /[aeiouAEIOU]/;
  let vowelRepeaterArr = str.split("");
  let vowelCounter = 0;

  vowelRepeaterArr = vowelRepeaterArr.map(element => {
    let vowelRepeaterElement = element;
    
    if (regex.test(element)) {
      for (let i = 0; i < vowelCounter; i ++) {
        vowelRepeaterElement += element.toLowerCase();
      }
    vowelCounter ++;
    }

    return vowelRepeaterElement;
  })

  return vowelRepeaterArr.join('');
}

console.log(repeatVowels("hello world")); 
console.log(repeatVowels("freeCodeCamp"));
console.log(repeatVowels("AEIOU"));
console.log(repeatVowels("I like eating ice cream in Iceland"));

File: 2025-09-04/vowel_repeater.py
import re

def repeat_vowels(s):
    regex = re.compile(r"[aeiouAEIOU]")
    vowel_repeater_str = ''
    vowel_counter = 0;

    for char in s:
        vowel_repeater_element = char
        if re.match(regex, char):
            vowel_repeater_element += char.lower() * vowel_counter
            vowel_counter += 1
        vowel_repeater_str += vowel_repeater_element
        
    return vowel_repeater_str

print(repeat_vowels("hello world")) 
print(repeat_vowels("freeCodeCamp"))
print(repeat_vowels("AEIOU"))
print(repeat_vowels("I like eating ice cream in Iceland"))